packages/codemodel.go/src/type.ts

```diff
@@ -359,54 +368,68 @@ export function getLiteralTypeDeclaration(literal: LiteralType): string {
 
 /**
  * returns the Go type declaration for the specified type.
- * any value in pkgName is prefixed to the underlying type name.
+ * if the type is defined in a package that's different from
+ * the provided scope, the type declaration will include
+ * the type's package name prefix.
  * 
  * @param type the type for which to emit the declaration
- * @param pkgName optional package name prefix for the type
+ * @param scope the scope in which the type declaration is emitted
  * @returns the Go type declaration
  */
-export function getTypeDeclaration(type: Type, pkgName?: string): string {
+export function getTypeDeclaration(type: Client | Type, scope: PackageType): string {
   switch (type.kind) {
     case 'any':
     case 'string':
       return type.kind;
+    case 'client':
     case 'clientOptions':
     case 'constant':
+    case 'constantValue':
     case 'interface':
     case 'model':
     case 'paramGroup':
     case 'polymorphicModel':
-      if (pkgName) {
-        return `${pkgName}.${type.name}`;
+    case 'responseEnvelope': {
+      let pkg: PackageType;
+      const typeName = type.kind === 'paramGroup' ? type.groupName : type.name;
+      switch (type.kind) {
+        case 'constantValue':
+          pkg = type.type.pkg;
+          break;
+        case 'responseEnvelope':
+          pkg = type.method.receiver.type.pkg;
```

Comment:
Isn't this always the case i.e., we never execute the `constantValue` case? I may be misreading/understanding but it appears we have
```ts
switch (type.kind) {
  case 'responseEnvelope':
    switch (type.kind) {
      case 'constantValue':
      case 'responseEnvelope':
```